src/diff/props.js

```diff
@@ -13,6 +13,11 @@ function setStyle(style, key, value) {
 	}
 }
 
+let globalCounter = Number.MIN_SAFE_INTEGER;
+const getCounter = increment => {
+	return increment ? ++globalCounter : globalCounter;
```

Comment:
I don't think it's safe to assume that the user has a certain amount of events, if we however memoize this like the example I gave in the issue that would become a lot safer as then the assumption becomes amount of re-renders instead of amount of events.

Reply:
maybe force globalCounter to be a 32 bit int using `globalCounter | 0`? as far as i know javascript VMs, it would be a tad bit faster.

additionally it would at least lead to predictable overflows whereas a regular double would start doing interesting things at around 2^52.

Reply:
Take a look at the update. Used the counter [technique mentioned in the issue](https://github.com/vuejs/core/blob/main/packages/runtime-dom/src/modules/events.ts#L82-L83), forced counters to be unsigned int with  >>> 0.

Reply:
do i miss something that makes `>>> 0` better than `| 0`? it's longer and halves the available range...

Reply:
I agree that it is uglier, but if I understand it correctly the shift operator converts a number to an  unsigned 32 bit int with the range of [0 to 4294967295]. The bitwise OR converts a number to a signed 32 bit int with the range of [-2147483648 to 2147483647]. 
When we start the counter with 0 the unsigned int doubles the range. Right?
You can check it with code like this:
```
const x = 0b1111111111111111111111111111111 >>> 0; // 2^31 -1 as unsigned int32
const y = 0b1111111111111111111111111111111 | 0; // 2^31 -1 as signed int32
console.log(x, y);
console.log(x + 1 >>> 0, y + 1 | 0);
```
I've read about the signed/unsigned manipulation in [this article](https://james.darpinian.com/blog/integer-math-in-javascript). I wonder if we want to use those manipulations at all?

Reply:
i think i misunderstood what `>>>` does and unsigned should have exactly the same range size as signed (both are 32 bits in the end...). i definitely would use integers instead of floating point, it just feels a lot better :)

Reply:
The thing bothering me most about this approach is when it rolls over (rare case) we might have event-handlers registered with MAX_INT_32 and 0 which all of a sudden would incorrectly bail out of a bubbled click 😅 thinking about alternatives like registering event-handlers during the `commit` phase but not sure as of just yet